src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,278 @@
+        public String key() {
+          return FIREBASE_ANALYTICS_KEY;
+        }
+      };
 
-    @Override public String key() {
-      return FIREBASE_ANALYTICS_KEY;
-    }
-  };
+  static final Pattern ISO_DATE =
+          Pattern.compile("(^([0-9]{4})-(1[0-2]|0[1-9])-(0[1-9]|1[1-9]|2[0-9]|3[0-1]).*)", CASE_INSENSITIVE);
+  private static final String FIREBASE_ANALYTICS_KEY = "Firebase";
+  final Logger logger;
+  final FirebaseAnalytics firebaseAnalytics;
+  private static final Map<String, String> eventMapper = createEventMap();
+  private static Map<String, String> createEventMap()
+  {
+    Map<String,String> eventMapper = new HashMap<>();
+    eventMapper.put("Product Added", Event.ADD_TO_CART);
+    eventMapper.put("Checkout Started", Event.BEGIN_CHECKOUT);
+    eventMapper.put("Order Completed", Event.ECOMMERCE_PURCHASE);
+    eventMapper.put("Order Refunded", Event.PURCHASE_REFUND);
+    eventMapper.put("Product Viewed", Event.VIEW_ITEM);
+    eventMapper.put("Product List Viewed", Event.VIEW_ITEM_LIST);
+    eventMapper.put("Payment Info Entered", Event.ADD_PAYMENT_INFO);
+    eventMapper.put("Promotion Viewed", Event.PRESENT_OFFER);
+    eventMapper.put("Product Added to Wishlist", Event.ADD_TO_WISHLIST);
+    eventMapper.put("Product Shared", Event.SHARE);
+    eventMapper.put("Product Clicked", Event.SELECT_CONTENT);
+    eventMapper.put("Product Searched", Event.SEARCH);
+    eventMapper.put("Promotion Viewed", Event.PRESENT_OFFER);
+    return eventMapper;
+  }
+  private static final Map<String, String> propertyMapper = createPropertyMap();
+  private static Map<String, String> createPropertyMap()
+  {
+    Map<String,String> propertyMapper = new HashMap<>();
+    propertyMapper.put("category", Param.ITEM_CATEGORY);
+    propertyMapper.put("product_id", Param.ITEM_ID);
+    propertyMapper.put("name", Param.ITEM_NAME);
+    propertyMapper.put("price", Param.PRICE);
+    propertyMapper.put("quantity", Param.QUANTITY);
+    propertyMapper.put("query", Param.SEARCH_TERM);
+    propertyMapper.put("shipping", Param.SHIPPING);
+    propertyMapper.put("tax", Param.TAX);
+    propertyMapper.put("total", Param.VALUE);
+    propertyMapper.put("revenue", Param.VALUE);
+    propertyMapper.put("order_id", Param.TRANSACTION_ID);
+    propertyMapper.put("currency", Param.CURRENCY);
+    return propertyMapper;
+  }
+
+  public FirebaseIntegration(Context context, Logger logger) {
+    this.logger = logger;
+    firebaseAnalytics = FirebaseAnalytics.getInstance(context);
+  }
+
+  @Override
+  public void onActivityResumed(Activity activity) {
+    super.onActivityResumed(activity);
 
-  private static final String FIREBASE_ANALYTICS_KEY = "Firebase Analytics";
+    PackageManager packageManager = activity.getPackageManager();
+    try {
+      ActivityInfo info =
+              packageManager.getActivityInfo(activity.getComponentName(), PackageManager.GET_META_DATA);
+      CharSequence activityLabel = info.loadLabel(packageManager);
```

Comment:
Instead of calling `toString` twice later, this could just be `String activityLabel = info.loadLabel(packageManager).toString();`